Fix sjoin with duplicate fieldnames

brahn opened the pull request:
See https://github.com/chrispahm/spatialmerge/issues/2

brahn committed:
don't read unused file (required to make tests pass)

brahn committed:
bugfix: read values from object2, not object1

brahn committed:
to avoid overwriting values, set defaults for lsuffix and rsuffix

brahn committed:
bugfix: find duplicate property keys when merging

chrispahm commented on test/test.js:
```diff
@@ -32,7 +33,7 @@ const readFile = util.promisify(fs.readFile)
   const mergeExpected = require('./output/merged.json')
   const citiesExpected = require('./output/merged_cities.json')
-  const plotsExpected = await reader('./test/output/merged_plots.json')
+  // const plotsExpected = await reader('./test/output/merged_plots.json')
```
Yes, this was an old test file, look like I forgot to delete the line accordingly

chrispahm commented on src/helpers.js:
```diff
@@ -21,13 +36,14 @@ function mergeWith (object1, object2, lsuffix, rsuffix) {
         feature[key + '_' + rsuffix] = object2[key]
       } else {
-        feature[key] = object1[key]
+        feature[key] = object2[key]
```
That's right, this has been a copy/paste bug

chrispahm reviewed (commented):
LGTM! However, we should add a test using the data from #2 to ensure that sjoin returns the same output as geopandas. I'll work on that!

chrispahm committed:
💄 rename object1 and object2 to left- and rightFeatureProps for better readability

chrispahm committed:
🐛 fix bug when cloining features with null geometries

chrispahm committed:
Update countries.csv

🐛 remove countries from test file that do not have a ISO_A3 code → fixes arbitrary merging results when comparing to GeoPandas

chrispahm committed:
Update countries.json

🐛 remove countries from test json file that do not have a ISO_A3 code → fixes arbitrary merging results when comparing to GeoPandas

chrispahm committed:
✅ add / update expected results from GeoPandas

 → see Jupyter notebook

chrispahm committed:
✅ add test for #3, ✔️ fix other tests

chrispahm committed:
🔥 remove unused test files

chrispahm committed:
🧹 remove writing sjoin results from test file

brahn commented on src/helpers.js:
```diff
@@ -14,29 +14,29 @@ function findAllDuplicates(array) {
-  const keys2 = Object.keys(object2)
+function mergeWith(leftFeatureProps, rightFeatureProps, lsuffix = 'left', rsuffix = 'right') {  
+  const keys1 = Object.keys(leftFeatureProps)
+  const keys2 = Object.keys(rightFeatureProps)
```
good call!

brahn commented on test/test.js:
```diff
@@ -8,7 +8,11 @@ import { strictEqual, deepStrictEqual } from 'assert'
+
+const sortBy = (fn) => {
+  return (a, b) => (fn(a) > fn(b)) ? 1 : ((fn(b) > fn(a)) ? -1 : 0);
+}
```
I tend to pull in https://underscorejs.org/ for this and other niceties but this works just fine :-)

brahn reviewed (commented):
Looks good to me!

chrispahm committed:
1.0.1
